app/controllers/api/v1/visualization/annotations_controller.rb

```diff
@@ -0,0 +1,180 @@
+module Api
+  module V1
+    module Visualization
+      # API methods for visualizing annotation-related data
+      # does NOT contain methods for editing annotations
+      class AnnotationsController < ApiBaseController
+        include Concerns::Authenticator
+        include Concerns::StudyAware
+        include Concerns::ApiCaching
+        include Swagger::Blocks
+
+        VALID_SCOPE_VALUES = ['study', 'cluster']
+        VALID_TYPE_VALUES = ['group', 'numeric']
+
+        before_action :set_current_api_user!
+        before_action :set_study
+        before_action :check_study_view_permission
+        before_action :check_api_cache!
+        after_action :write_api_cache!
+
+        annotation_description_doc = 'Object with name (String), values (Array or uique values), type (String), scope (String), and cluster_name (string, if applicable)'
+
+        swagger_path '/studies/{accession}/annotations' do
+          operation :get do
+            key :tags, [
+                'Visualization'
+            ]
+            key :summary, 'Get all annotations for the study'
+            key :description, 'Get all annotations for the study, with name, values, type, scope, and cluster_name if applicable'
+            key :operationId, 'study_annotations_path'
+            parameter({
+              name: :accession,
+              in: :path,
+              description: 'Study accession number (e.g. SCPXXX)',
+              required: true,
+              type: :string
+            })
+            response 200 do
+              key :description, 'Array of Annotation objects'
+              schema do
+                key :type, :array
+                key :title, 'Array'
+                items do
+                  key :title, 'Annotation'
+                  key :description, annotation_description_doc
+                end
+              end
+            end
+            extend SwaggerResponses::StudyControllerResponses
+          end
+        end
+
+        #
+        def index
+          render json: ClusterVizService.available_annotations(@study, nil, current_api_user)
```

Comment:
Oh man, you hit on the issue that caused me the most waffling on this entire PR.  I tried a few different configs of clust/annot controllers/services, and they were all vaguely unsatisfying.  I can't even say this current config is the least bad, it just happened to be the one at which I finished the PR. The core problem is that it will never be really obvious which service a given method belongs in.  e.g. if I'm grabbing a list of all annotations of a cluster, is that a cluster or annotation service method? Either way, the controllers will be calling methods in the other service (e.g. get annotation will always need to call a get_cluster method which belongs in the ClusterVizService).  The good news is that it's really easy to rearrange methods between service classes, so I'd propose we table this until we have a few more of these controllers and can see what patterns emerge

Reply:
I see the issue, and I definitely wondered about some methods as a read them.  You could just move everything annotation-specific into `AnnotationVizService`, except methods that are definitively cluster-specific.  

Strawman for `AnnotationVizService`

- self.load_cluster_group_annotations(study, cluster, current_user) -- from `ClusterVizService`
- self.get_study_annotation_options(study, user) -- from `ClusterVizService`
- self.available_annotations(study, cluster, current_user, annotation_type=nil) -- from `ClusterVizService`
- self.annotation_cell_values_tsv(study, cluster, annotation) -- from `ClusterVizService`
- def self.parse_annotation_legacy_params(study, params) -- from `ExpressionVizService`
- def self.get_selected_annotation(study, cluster, annot_name, annot_type, annot_scope) -- from `ExpressionVizService`

It's not perfect, but it feels to me like those are definitely annotation-based, as opposed to cluster- or expression-based.  I could also see `annotation_cell_values_tsv` staying in `ClusterVizService` as well.